### DISCUSSION TITLE
Restarting qtile moves the current workspace to another and changes the current layout

### DISCUSSION THREAD

**Discussion by ghost** [qtile/qtile#2302] [Q&A]
i have 5 workspaces - 1 to 5. I opened 2 terminals and a browser and on **workspace 1**. And changed the layout to **max**

Pressed <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>R</kbd> to restart. Now all windows switched to workspace 4 and layout switched to columns. Restarted again, all windows switched to workspace 5

**Comment by m-col** [comment]
I use something like this:
```py
@hook.subscribe.startup
def _initial_groups():
    if len(qtile.screens) > 1:
        qtile.groups_map['q'].cmd_toscreen(1, toggle=False)
    qtile.groups_map['1'].cmd_toscreen(0, toggle=False)
```
Which sets my default groups to `'1'` and if I have a second screen attached also `'q'`.

If you are restarting in order to re-configure screens then as of https://github.com/qtile/qtile/commit/9e6af506cd714fc993c5b7f534c689b80cc21793 this is no longer necessary. Once we can reload the config with a dedicated `cmd_` (https://github.com/qtile/qtile/issues/2053) then restarting will not be necessary at all except to update to a new Qtile version.

**Reply by ghost** [reply]
I have just 1 monitor, and I restart after editing the config
